app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -276,11 +319,62 @@ class PipelineViz extends React.Component {
     });
   }
 
-  handleNodeHover(stageIndex, info) {
-    const { highlightColor } = this.props;
+  shouldUpdateMouseMove(x, y) {
+    const { minMouseMoveUpdateDistance } = this.props;
+    const distance = Math.sqrt(
+      Math.pow(x - this.lastMouseMoveInfo.x, 2) +
+        Math.pow(y - this.lastMouseMoveInfo.y, 2)
+    );
+    return distance >= minMouseMoveUpdateDistance;
+  }
+
+  handleMouseMove(stageIndex, e) {
+    const graph = this.graphs[stageIndex];
+    if (!this.shouldUpdateMouseMove(e.clientX, e.clientY)) {
+      return;
+    }
+
+    const rect = e.target.getBoundingClientRect();
+    const nodeId = this.getNodeIdAtCoords(
+      graph,
+      e.clientX - rect.left,
+      e.clientY - rect.top
+    );
+
+    if (
+      stageIndex != this.lastMouseMoveInfo.graphIndex ||
+      nodeId != this.lastMouseMoveInfo.nodeId
+    ) {
+      this.handleNodeBlur();
+    }
+
+    Object.assign(this.lastMouseMoveInfo, {
+      graphIndex: stageIndex,
+      nodeId: nodeId,
```

Comment:
you could just do `nodeId`

Reply:
`handleNodeBlur` doesn't update `lastMouseMoveInfo` and I want to keep that method to be clearing previous styling (as opposed to updating to the current values)